File: 405 Convert a Number to Hexadecimal/405.js
/**
 * @param {number} num
 * @return {string}
 */
var toHex = function(num) {
    if (num === 0) {
        return '0';
    }
    let map = "abcdef"
    let res = []
    while (num !== 0) {
        let hexNum = num  & 0xF
        res.push(hexNum < 10 ? hexNum : map[hexNum - 10])
        num >>>= 4
    }

    return res.reverse().join('')
};

File: 405 Convert a Number to Hexadecimal/405.py
class Solution:
    def toHex(self, num: int) -> str:
        if not num:
            return '0'
        map = "0123456789abcdef"
        res = []
        zeroFlag = False
        for i in range(28, -4, -4):
            hexNum = (num >> i) & 0xF
            if hexNum != 0 or zeroFlag:
                res.append(map[hexNum])
                zeroFlag = True
        return ''.join(res)

    def toHex2(self, num: int) -> str:
        if not num:
            return '0'
        map = "0123456789abcdef"
        if num < 0:
            num += 0x100000000
        res = []
        while num:
            hexNum = num & 0xF
            res.append(map[hexNum])
            num >>= 4
        return ''.join(res[::-1])